controller/program/src/instructions/start_rebalance.rs

```diff
@@ -0,0 +1,491 @@
+        },
+        out_lst_idx,
+    )?;
+
+    lst_sync_sol_val_unchecked(
+        abr,
+        cpi,
+        SyncSolValueIxAccs {
+            ix_prefix: NewSyncSolValueIxPreAccsBuilder::start()
+                .with_lst_mint(*ix_prefix.inp_lst_mint())
+                .with_pool_state(*ix_prefix.pool_state())
+                .with_lst_state_list(*ix_prefix.lst_state_list())
+                .with_pool_reserves(*ix_prefix.inp_pool_reserves())
+                .build(),
+            calc_prog: inp_calc_prog,
+            calc: inp_calc,
+        },
+        inp_lst_idx,
+    )?;
+
+    let old_total_sol_value = {
+        let pool = unsafe { PoolState::of_acc_data(abr.get(*ix_prefix.pool_state()).data()) }
+            .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidPoolStateData))?;
+        pool.total_sol_value
+    };
+
+    // Transfer out_lst tokens from reserves to withdraw_to account
+    let out_lst_mint_data = abr.get(*ix_prefix.out_lst_mint()).data();
+    let out_lst_mint = RawMint::of_acc_data(out_lst_mint_data)
+        .and_then(Mint::try_from_raw)
+        .ok_or(INVALID_ACCOUNT_DATA)?;
+    let decimals = out_lst_mint.decimals();
+
+    let transfer_checked_ix_data = TransferCheckedIxData::new(args.amount, decimals);
+    let transfer_checked_accs = NewTransferCheckedIxAccsBuilder::start()
+        .with_src(*ix_prefix.out_pool_reserves())
+        .with_mint(*ix_prefix.out_lst_mint())
+        .with_dst(*ix_prefix.withdraw_to())
+        .with_auth(*ix_prefix.pool_state())
+        .build();
+    let out_lst_token_program_key = *abr.get(*ix_prefix.out_lst_token_program()).key();
+
+    cpi.invoke_signed(
+        abr,
+        &out_lst_token_program_key,
+        transfer_checked_ix_data.as_buf(),
+        transfer_checked_ix_account_handle_perms(transfer_checked_accs),
+        &[PdaSigner::new(&[
+            PdaSeed::new(b"state"),
+            PdaSeed::new(&[POOL_STATE_BUMP]),
+        ])],
+    )?;
+
+    lst_sync_sol_val_unchecked(
+        abr,
+        cpi,
+        SyncSolValueIxAccs {
+            ix_prefix: NewSyncSolValueIxPreAccsBuilder::start()
+                .with_lst_mint(*ix_prefix.out_lst_mint())
+                .with_pool_state(*ix_prefix.pool_state())
+                .with_lst_state_list(*ix_prefix.lst_state_list())
+                .with_pool_reserves(*ix_prefix.out_pool_reserves())
+                .build(),
+            calc_prog: out_calc_prog,
+            calc: out_calc,
+        },
+        out_lst_idx,
+    )?;
+
+    let rebalance_record_space = size_of::<RebalanceRecord>();
+
+    abr.get_mut(*ix_prefix.rebalance_record())
+        .realloc(rebalance_record_space, false)?;
+
+    let rebalance_record_seeds = [
+        PdaSeed::new(b"rebalance-record"),
+        PdaSeed::new(&[inf1_ctl_jiminy::keys::REBALANCE_RECORD_BUMP]),
+    ];
+    let rebalance_record_signer = PdaSigner::new(&rebalance_record_seeds);
+
+    let system_prog_key = *abr.get(*ix_prefix.system_program()).key();
```

Comment:
nit: for system program, lets just always use the hardcoded `SYS_PROG_ID` for better safety